### ISSUE TITLE
Can't upload `ipa` in Fabric with new Crashlytics iOS SDK

### ISSUE THREAD

**Issue by SergeMaslyakov** [fastlane/fastlane#12820]
Hi.

I have update 'Crashlytics-SDK' for iOS. After that `fastlane` couldn't upload `ipa`.

Error message with empty exit status
`Shell command exited with exit status  instead of 0.`

`Crashlytics.framework/submit 1.3.5 (18)`
`Fabric.framework/run 1.7.0 (208)`

With previous version of SDK all works fine.

Logs are here:

```
[12:51:05]	[15:51:05]: Get started using a Gemfile for fastlane https://docs.fastlane.tools/getting-started/ios/setup/#use-a-gemfile
DEBUG [2018-06-29 15:51:06.10]: Checking if there are any plugins that should be loaded...
INFO [2018-06-29 15:51:06.25]: ----------------------------------------
INFO [2018-06-29 15:51:06.25]: --- Step: Verifying fastlane version ---
INFO [2018-06-29 15:51:06.25]: ----------------------------------------
INFO [2018-06-29 15:51:06.25]: Your fastlane version 2.98.0 matches the minimum requirement of 2.70.1  ✅
INFO [2018-06-29 15:51:06.26]: ------------------------------
INFO [2018-06-29 15:51:06.26]: --- Step: default_platform ---
INFO [2018-06-29 15:51:06.26]: ------------------------------
INFO [2018-06-29 15:51:06.26]: Driving the lane 'beta'   
INFO [2018-06-29 15:51:06.26]: ----------------------------------------
INFO [2018-06-29 15:51:06.26]: --- Step: changelog_from_git_commits ---
INFO [2018-06-29 15:51:06.26]: ----------------------------------------
INFO [2018-06-29 15:51:06.26]: Collecting the last 13 Git commits

                        -- git commits - redacted ----

INFO [2018-06-29 15:51:06.28]: -------------------------
INFO [2018-06-29 15:51:06.28]: --- Step: crashlytics ---
INFO [2018-06-29 15:51:06.28]: -------------------------
ERROR [2018-06-29 15:51:06.28]: Overwriting :notes_path, because you specified :notes
INFO [2018-06-29 15:51:06.29]: Uploading the build to Crashlytics Beta. Time for some ☕️.
DEBUG [2018-06-29 15:51:06.29]: ./Core/ThirdParty/Crashlytics.framework/submit [[API_TOKEN]] [[BUILD_SECRET]] -ipaPath 'xxx.ipa' -emails 'xxx@xxxxxx.com' -notesPath 'xxxxxx/changelog20180629-76488-1kbr4u7' -groupAliases 'xxxxx-team' -notifications YES -debug NO
ERROR [2018-06-29 15:51:18.95]: Shell command exited with exit status  instead of 0.
WARN [2018-06-29 15:51:18.95]: Lane Context:
INFO [2018-06-29 15:51:18.95]: {:DEFAULT_PLATFORM=>:ios, :PLATFORM_NAME=>nil, :LANE_NAME=>"beta", :FL_CHANGELOG=>"xxxxxx"}
ERROR [2018-06-29 15:51:18.95]: 2018-06-29 15:51:06.314 submit Crashlytics: Crashlytics.framework/submit 1.3.5 (18)

INFO [2018-06-29 15:51:18.96]: Successfully generated documentation at path '/tc/ff0923c7705629b0/fastlane/README.md'

+------+----------------------------+-------------+
|                fastlane summary                 |
+------+----------------------------+-------------+
| Step | Action                     | Time (in s) |
+------+----------------------------+-------------+
| 1    | Verifying fastlane version | 0           |
| 2    | default_platform           | 0           |
| 3    | changelog_from_git_commits | 0           |
|      | crashlytics                | 12          |
/xxxxx/.rbenv/versions/2.4.0/lib/ruby/gems/2.4.0/gems/fastlane-2.98.0/fastlane_core/lib/fastlane_core/ui/interface.rb:141:in `user_error!': [!] 2018-06-29 15:51:06.314 submit Crashlytics: Crashlytics.framework/submit 1.3.5 (18) (FastlaneCore::Interface::FastlaneError)
+------+----------------------------+-------------+


ERROR [2018-06-29 15:51:19.01]: fastlane finished with errors
	from /xxxx/.rbenv/versions/2.4.0/lib/ruby/gems/2.4.0/gems/fastlane-2.98.0/fastlane_core/lib/fastlane_core/ui/ui.rb:17:in `method_missing'
	from /xxxx/.rbenv/versions/2.4.0/lib/ruby/gems/2.4.0/gems/fastlane-2.98.0/fastlane/lib/fastlane/actions/crashlytics.rb:42:in `block in run'
	from /xxxx/.rbenv/versions/2.4.0/lib/ruby/gems/2.4.0/gems/fastlane-2.98.0/fastlane/lib/fastlane/helper/sh_helper.rb:78:in `sh_control_output'
	from /xxxx/.rbenv/versions/2.4.0/lib/ruby/gems/2.4.0/gems/fastlane-2.98.0/fastlane/lib/fastlane/actions/crashlytics.rb:45:in `run'
	from /xxxx/.rbenv/versions/2.4.0/lib/ruby/gems/2.4.0/gems/fastlane-2.98.0/fastlane/lib/fastlane/runner.rb:257:in `block (2 levels) in execute_action'
	from /xxxx/.rbenv/versions/2.4.0/lib/ruby/gems/2.4.0/gems/fastlane-2.98.0/fastlane/lib/fastlane/actions/actions_helper.rb:50:in `execute_action'
	from /xxxx/.rbenv/versions/2.4.0/lib/ruby/gems/2.4.0/gems/fastlane-2.98.0/fastlane/lib/fastlane/runner.rb:236:in `block in execute_action'
	from /xxxx/.rbenv/versions/2.4.0/lib/ruby/gems/2.4.0/gems/fastlane-2.98.0/fastlane/lib/fastlane/runner.rb:231:in `chdir'
	from /xxxx/.rbenv/versions/2.4.0/lib/ruby/gems/2.4.0/gems/fastlane-2.98.0/fastlane/lib/fastlane/runner.rb:231:in `execute_action'
	from /xxxx/.rbenv/versions/2.4.0/lib/ruby/gems/2.4.0/gems/fastlane-2.98.0/fastlane/lib/fastlane/runner.rb:157:in `trigger_action_by_name'
	from /xxxx/.rbenv/versions/2.4.0/lib/ruby/gems/2.4.0/gems/fastlane-2.98.0/fastlane/lib/fastlane/fast_file.rb:149:in `method_missing'
	from Fastfile:13:in `block in parsing_binding'
	from /xxxx/.rbenv/versions/2.4.0/lib/ruby/gems/2.4.0/gems/fastlane-2.98.0/fastlane/lib/fastlane/lane.rb:33:in `call'
	from /xxxx/.rbenv/versions/2.4.0/lib/ruby/gems/2.4.0/gems/fastlane-2.98.0/fastlane/lib/fastlane/runner.rb:49:in `block in execute'
	from /xxxx/.rbenv/versions/2.4.0/lib/ruby/gems/2.4.0/gems/fastlane-2.98.0/fastlane/lib/fastlane/runner.rb:45:in `chdir'
	from /xxxx/.rbenv/versions/2.4.0/lib/ruby/gems/2.4.0/gems/fastlane-2.98.0/fastlane/lib/fastlane/runner.rb:45:in `execute'
	from /xxxx/.rbenv/versions/2.4.0/lib/ruby/gems/2.4.0/gems/fastlane-2.98.0/fastlane/lib/fastlane/lane_manager.rb:59:in `cruise_lane'
	from /xxxx/.rbenv/versions/2.4.0/lib/ruby/gems/2.4.0/gems/fastlane-2.98.0/fastlane/lib/fastlane/command_line_handler.rb:36:in `handle'
	from /xxxx/.rbenv/versions/2.4.0/lib/ruby/gems/2.4.0/gems/fastlane-2.98.0/fastlane/lib/fastlane/commands_generator.rb:107:in `block (2 levels) in run'
	from /xxxx/.rbenv/versions/2.4.0/lib/ruby/gems/2.4.0/gems/commander-fastlane-4.4.6/lib/commander/command.rb:178:in `call'
	from /xxxx/.rbenv/versions/2.4.0/lib/ruby/gems/2.4.0/gems/commander-fastlane-4.4.6/lib/commander/command.rb:153:in `run'
	from /xxxx/.rbenv/versions/2.4.0/lib/ruby/gems/2.4.0/gems/commander-fastlane-4.4.6/lib/commander/runner.rb:476:in `run_active_command'
	from /xxxx/.rbenv/versions/2.4.0/lib/ruby/gems/2.4.0/gems/fastlane-2.98.0/fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb:75:in `run!'
	from /xxxx/.rbenv/versions/2.4.0/lib/ruby/gems/2.4.0/gems/commander-fastlane-4.4.6/lib/commander/delegates.rb:15:in `run!'
	from /xxxx/.rbenv/versions/2.4.0/lib/ruby/gems/2.4.0/gems/fastlane-2.98.0/fastlane/lib/fastlane/commands_generator.rb:332:in `run'
	from /xxxx/.rbenv/versions/2.4.0/lib/ruby/gems/2.4.0/gems/fastlane-2.98.0/fastlane/lib/fastlane/commands_generator.rb:41:in `start'
	from /xxxx/.rbenv/versions/2.4.0/lib/ruby/gems/2.4.0/gems/fastlane-2.98.0/fastlane/lib/fastlane/cli_tools_distributor.rb:108:in `take_off'
	from /xxxx/.rbenv/versions/2.4.0/lib/ruby/gems/2.4.0/gems/fastlane-2.98.0/bin/fastlane:20:in `<top (required)>'
	from /xxxx/.rbenv/versions/2.4.0/bin/fastlane:23:in `load'
	from /xxxx/.rbenv/versions/2.4.0/bin/fastlane:23:in `<main>'
Process exited with code 1
Process exited with code 1
Step Upload to fabric (Command Line) failed
```

**Comment by rlester** [comment]
I am seeing this as well. Same SDK versions.

**Comment by spaquet** [comment]
I have the same issue
Same version
Fastlane v 2.98.0
Ruby v 2.4.3p205
Xcode : 9.4.1 (9F2000)

**Comment by spaquet** [comment]
For ow the only solution I found is to downgrade to version 3.10.2 which is working perfectly fine.

**Comment by filmhomage** [comment]
Same problem. 

> For ow the only solution I found is to downgrade to version 3.10.2 which is working perfectly fine.

Current solution. 👍🏻

**Comment by slxl** [comment]
have the same problem.

guys, @filmhomage & @spaquet - _what_ should I downgrade to 3.10.2? :) isn't obvious from context

**Comment by vovkaprigarin** [comment]
Hm... path to submit script is incorrect. Need use` {PODS_DIR}/Crashlytics/submit` instead `{PODS_DIR}/Crashlytics/iOS/Crashlytics.framework/submit`. Fix it pls:)

**Comment by spaquet** [comment]
@slxl : just updated my comment to make it easier to understand.

**Comment by spaquet** [comment]
@svilon I have no notes or notes_path in my fastlane script and nothing works when I'm using Crashlytics v 3.10.3, while all previous versions are working like a charm

**Comment by zapjonny** [comment]
When uploading builds through my script, since today I get 

> submit Crashlytics: Crashlytics.framework/submit 1.3.5 (18)


Not sure if same, but will also try downgrading. Just upgraded to latest crashlytics today.

**Comment by zapjonny** [comment]
Btw so it's a bug in Crashlytics SDK? I think we're in the wrong issue tracker :-)

**Comment by vovkaprigarin** [comment]
@zapjonny no, fastlane use incorrect path to submit script.

**Comment by Morse-Code** [comment]
@zapjonny can you confirm fixing the path resolves the issue? @svilon indicated that this does not work.

**Comment by zapjonny** [comment]
Ok I will have to confirm. Going back to 3.10.2 and I could submit again.

**Comment by zapjonny** [comment]
Btw things stopped working after upgrading Crashlytics and not fastlane so kind of feels like a problem with crashlytics but anywhoo

**Comment by Morse-Code** [comment]
@zapjonny agreed.

**Comment by Morse-Code** [comment]
Interestingly, nothing regarding installation paths changed between my projects previous Crashlytics installation (3.9.3) and 3.10.3. So, I doubt the path is incorrect in Fastlane. Also, if you notice there is a submit executable at both `{PODS_DIR}/Crashlytics/iOS/Crashlytics.framework/submit` & `{PODS_DIR}/Crashlytics/submit`. If you run a diff on these two files you'll see they are actually the same.

**Comment by zapjonny** [comment]
I find three submit binaries

Crashlytics/submit
Crashlytics/Crashlytics.framework/submit
Crashlytics/iOS/Crashlytics.framework/submit

But as you say it doesn't look like these paths changed since previous version.

**Comment by zapjonny** [comment]
Next to 
Crashlytics/Crashlytics.framework/submit
there is a
Crashlytics/Crashlytics.framework/README
saying

We've now combined all our supported platforms into a single podspec. As a result, we moved our submit script to a new location for Cocoapods projects: ${PODS_ROOT}/Crashlytics/submit. To avoid breaking functionality that references the old location of the submit, we've placed this dummy script that calls to the correct location, while providing a helpful warning if it is invoked. This bridge for backwards compatibility will be removed in a future release, so please heed the warning!


I have never looked into the inside of this framework so can't tell if this is new information or not.

**Comment by vovkaprigarin** [comment]
@Morse-Code yes, try manually use submit script from .framework and from directory of Crashlytics. You will catch a error in the first case.

**Comment by zapjonny** [comment]
Yes, tried this, and @svilon is right, sending notes and

```
2018-07-03 00:49:45.528 submit Crashlytics: Crashlytics.framework/submit 1.3.5 (18)
Segmentation fault: 11

```
if I remove notes, submit succeeds.

**Comment by zapjonny** [comment]
No but if Crashlytics/submit segfaults then the problem is in Crashlytics... this issue could be closed...

**Comment by vovkaprigarin** [comment]
@zapjonny Fastlane uses incorrect path to submit script. Need fix this. Segfaults in Crashlytics its other issue.

**Comment by zapjonny** [comment]
Tried using

Fastlane 2.99.0
Crashlytics 3.10.3

got

[11:47:00]: fastlane finished with errors
[!] 2018-07-03 11:46:54.243 submit Crashlytics: Crashlytics.framework/submit 1.3.5 (18)

**Comment by vovkaprigarin** [comment]
@zapjonny because used `submit Crashlytics: **Crashlytics.framework/submit**`

**Comment by zapjonny** [comment]
This still segfaults

```
  desc "Temp testsendonly"
  lane :testsendonly do |values|
    emails = ['j_bergstrom@zappallas.com']
    crashlytics(api_token: CRASHLYTICS_APITOKEN,
             build_secret: CRASHLYTICS_BUILDSECRET,
         crashlytics_path: './Pods/Crashlytics',
                   emails: emails,
                    notes: 'testsendonly',
            notifications: true)
  end
```

**Comment by svilon** [comment]
Confirm, issue persists

**Comment by zapjonny** [comment]
Using default submit path and commenting out notes also works. So even though the Crashlytics action of Fastlane might use an old path (I haven't looked into the default path) it still works as long as notes is not setup and the big problem at this point is the submit binary of Crashlytics.

**Comment by vovkaprigarin** [comment]
default path in crashlytics_helper.rb : `path = Dir["./Pods/iOS/Crashlytics/Crashlytics.framework"].last || Dir["./**/Crashlytics.framework"].last
`

@zapjonny please, read my comments above.

**Comment by zapjonny** [comment]
Ok looks incorrect 🤔

**Comment by svilon** [comment]
@vovkaprigarin, you are generally right that paths should be updated. But it DOESN'T help with this issue. Maybe it's better to create another issue on this?

Also, not sure where you've found crashlytics_helper.rb. Any way, it should have contained `Pods/Crashlytics/iOS/Crashlytics.framework` as first path.
Also, when I run crashlytics action with `crashlytics_path:'./Pods/Crashlytics/Crashlytics.framework/'` I get graceful error:
`DEBUG [2018-07-03 05:59:17.97]: error: The submit binary delivered by cocoapods is in a new location, under '${PODS_ROOT}/Crashlytics/submit'. This script was put in place for backwards compatibility, but it relies on PODS_ROOT, which does not have a value in your current setup. Please update the path to the submit binary to fix this issue.`
Funny thing about it, in last case crashlytics finishes with success, error message is shown only in debug mode, but nothing is actually uploaded.

**Comment by Den-Ree** [comment]
@svilon we have the same 

_Funny thing about it, in last case crashlytics finishes with success, error message is shown only in debug mode, but nothing is actually uploaded._

**Comment by sprint84** [comment]
I'm experiencing the same issue with Crashlytics 3.10.1 and fastlane 2.98.0/2.99.0. So I believe it's some issue on Crashlytics side.

**Comment by spaquet** [comment]
So far Crashlytics 3.10.2 is reported as being working for fastlane 2.98.0 and 2.99.0

**Comment by Panajev** [comment]
We do not use fastlane's crashlytics action, the segfault in Crashlytics only occurs if you update to Crashlytics 3.10.3: https://stackoverflow.com/questions/51090226/crashlytics-3-10-3-cocoapods-submit-segmentation-fault-11

**Comment by rickvdl** [comment]
It at least seems to work with version `3.10.0`, so I downgraded. Download link: `https://s3.amazonaws.com/kits-crashlytics-com/ios/com.twitter.crashlytics.ios/3.10.0/com.crashlytics.ios-manual.zip`

**Comment by KrauseFx** [comment]
Thanks for reporting this, we're taking a look

**Comment by joshdholtz** [comment]
I did some further digging into this last night and this is the conclusion:

TL;DR: We have a fix for making sure bumping up the priority in finding the correct binary in PR #12876 and we are in contact with Crashlytics about pushing a fix for the the `submit` binary segfaulting when trying to attach notes

## Part 1: Moved `submit` binary (This is the part _fastlane_ can fix)

The Crashlytics pod actually had three `submit` binaries. Two of them could successfully run (see Part 2) and one would fail with an error.
- ✅ `${PODS_ROOT}/Crashlytics/submit`
  - This is the one that Crashlytics suggests users use but _fastlane_ was not looking at this one yet
  - PR #12876 will fix this
- ✅ `${PODS_ROOT}/Crashlytics/iOS/Crashlytics.framework/submit`
  - This is the binary that _fastlane_ was able to find and did use most of the time
  - This binary could successfully upload an IPA in version `3.10.3` and `3.10.4` but **only if** notes were not trying to be set (see Part 2)
- ❌ `${PODS_ROOT}/Crashlytics/Crashlytics.framework/submit`
  - There is also a binary at this location but executing this binary results in the following error
    - `error: The submit binary delivered by cocoapods is in a new location, under '${PODS_ROOT}/Crashlytics/submit'. This script was put in place for backwards compatibility, but it relies on PODS_ROOT, which does not have a value in your current setup. Please update the path to the submit binary to fix this issue`
  - _fastlane_ could have ended up potentially pointing to this location if `${PODS_ROOT}/Crashlytics/iOS/Crashlytics.framework/submit` wasn't found or if a _fastlane_ user explicitly pointed to this binary
  - This binary cannot be used at all

## Part 2: Broken binary when setting `:notes` and `:notes_path`

This part of the issue was caused by upgrading to Crashlytics version `3.10.3` and `3.10.4`. The `submit` binary located at `${PODS_ROOT}/Crashlytics/submit` and `${PODS_ROOT}/Crashlytics/iOS/Crashlytics.framework/submit`. Both of these binaries segfaulted when trying to attach `:notes` and `:notes_path` via the _fastlane_ action.

This segfault also occurred when attempting to execute both `${PODS_ROOT}/Crashlytics/submit` and `${PODS_ROOT}/Crashlytics/iOS/Crashlytics.framework/submit` directly via the command line (outside of _fastlane_) with the `-notesPath` argument.

Both of these above binaries succeeded in uploading a binary **if and only if** `-notesPath` argument was left out.

**Comment by vovkaprigarin** [comment]
> Set :crashlytics_path to ./Pods/Crashlytics

it's not work, because fastlane crashlytics_helper adds Crashlytics.framework/submit.
proof, line 8:
https://github.com/fastlane/fastlane/blob/master/fastlane/lib/fastlane/helper/crashlytics_helper.rb

**Comment by joshdholtz** [comment]
@vovkaprigarin Whoops, meant to say to use my branch/master or that one 😝

**Comment by KrauseFx** [comment]
Hey everyone, Crashlytics 3.10.5 was just shipped, make sure to update both _fastlane_ and Crashlytics to the most recent version and let us know if that works :+1:

**Comment by jsantana-broadsoft** [comment]
3.10.5 fixed our build (with a not updated fastlane). Thanks!

**Comment by svilon** [comment]
Works now, thanks for your help

**Comment by zapjonny** [comment]
submit didn't segfault.

Tried it twice.

Crashlytics didn't send out emails (to my account).

**Comment by spaquet** [comment]
@zapjonny emails is not stable, but that not new.
I already had discussions with guys at Fastlane and Crashlytics about this topic. If you think that there is an issue specific to that build, please open another ticket about it. This one is now closed and solved.
Thx
